### ISSUE TITLE
Need Group Block

### ISSUE THREAD

**Issue by itszarnish** [siyuan-note/siyuan#12984]
### In what scenarios do you need this feature?

In SiYuan, we should have something similar to groups in Capacities. We could create a group block and place content inside it. The benefit would be having a nicely rounded corner group, with the ability to change the background to any color. Please consider doing something about this.

Currently, there is no way to achieve this behavior in SiYuan, based on what I have explored.

Check out the attached video for how this feature works in Capacities:

https://github.com/user-attachments/assets/5f1e31f5-7886-444d-a23e-c475ae2b2b05

### Describe the optimal solution

-

**Comment by TCOTC** [comment]
It's similar to SiYuan's Super Block:

![image](https://github.com/user-attachments/assets/a5f381fe-8384-4c9c-b867-3179ee714d5a)

**Comment by itszarnish** [comment]
I tested this, and yes, it is quite similar to the group feature in Capacities. However, I’ve noticed what might be a bug that could be improved: there isn’t enough space on the left side of the super block.

![image](https://github.com/user-attachments/assets/66d61c1c-abd2-44a9-8494-ca6def17bbcb)

**Comment by itszarnish** [comment]
> I tested this, and yes, it is quite similar to the group feature in Capacities. However, I’ve noticed what might be a bug that could be improved: there isn’t enough space on the left side of the super block.
> 
> The spacing on the top, right and bottom seems right to me but left is a bit off.
> 
> **I think if the user hasn't applied any background color, it looks okay. However, once a background color is applied, there should be space on the left to match the top, right, and bottom spaces, so the content looks more balanced. I believe this can be improved.**
> 
> ![image](https://private-user-images.githubusercontent.com/119295635/381957721-66d61c1c-abd2-44a9-8494-ca6def17bbcb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzODA2NzIsIm5iZiI6MTczMDM4MDM3MiwicGF0aCI6Ii8xMTkyOTU2MzUvMzgxOTU3NzIxLTY2ZDYxYzFjLWFiZDItNDRhOS04NDk0LWNhNmRlZjE3YmJjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMVQxMzEyNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OWYyMjgxOTM3YTQyMTc3MWZkMDI3NTBlYmVlMGM1MTA2ZjQxODA1N2Y2YTVjYjI5ZjJkNjk5OWVmMDE5YTEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.zxvE3DV3swdF2lgCpy_VXy4SDmGo0wrFs9CbB2x2eIU)

It would be nice if we could apply borders to blocks as well. For example, in this super block case, I’ve applied a background color, but there should also be an option to apply a border.

This is exactly what I'm looking for:

![image](https://github.com/user-attachments/assets/21d9643a-b2da-4ecf-a8c9-8a9f88d9ccf9)

**Comment by TCOTC** [comment]
You might like the style of this super block:

```css
.protyle-wysiwyg [data-node-id].sb {
    padding: 4px 6px 4px 6px;
	background-color: var(--b3-table-even-background);
	box-shadow: inset 0 0 0 .6px var(--b3-border-color);
}
```

![image](https://github.com/user-attachments/assets/35fd6a49-cc76-45b7-8397-37f2cc4badfb)

How to use:

![image](https://github.com/user-attachments/assets/0f049d59-7427-4995-8b54-eef7e423ae45)

**Comment by itszarnish** [comment]
> You might like the style of this super block:
> 
> ```css
> .protyle-wysiwyg [data-node-id].sb {
>     padding: 4px 6px 4px 6px;
> 	background-color: var(--b3-table-even-background);
> 	box-shadow: inset 0 0 0 .6px var(--b3-border-color);
> }
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/78434827/381961234-35fd6a49-cc76-45b7-8397-37f2cc4badfb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzODEyMzIsIm5iZiI6MTczMDM4MDkzMiwicGF0aCI6Ii83ODQzNDgyNy8zODE5NjEyMzQtMzVmZDZhNDktY2M3Ni00NWI3LTgzOTctMzdmMmNjNGJhZGZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMxVDEzMjIxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzZjhhM2ZmNzY4NGEyZWJmZTRlZGU0NTkzNTM3NmZmNTRiODlkNTE4NDU0NzcxOTIwNGQzMzY3ZjY2ZTdiY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zvkj9orwWsMZjbAGnGERAWRaNpZcjHiPdwJedKHmrkU)
> 
> How to use:
> 
> ![image](https://private-user-images.githubusercontent.com/78434827/381961791-0f049d59-7427-4995-8b54-eef7e423ae45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzODEyMzIsIm5iZiI6MTczMDM4MDkzMiwicGF0aCI6Ii83ODQzNDgyNy8zODE5NjE3OTEtMGYwNDlkNTktNzQyNy00OTk1LThiNTQtZWVmN2U0MjNhZTQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMxVDEzMjIxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzNDlhZGRiNzliN2EzYzQwMGE0ODU0ZWU4ODExZGE1MzUyN2QxNGEwY2Q4MTgyM2U0NmE1YjZlY2E0OTA3YTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1FoJTH2kCsUPm3_c9Pwm5VPDmpZSbECEFUtAKmgpUxc)


Yes, I know about Code Snippets, but the problem is that I don’t want to apply this specific style to all blocks. I only want to apply borders to some blocks, not all of them, and Code Snippets can’t solve this issue. Instead, if the developers could implement a border feature, that would be great. 

Additionally, regarding the left space: once the user applies a background color, there should be some space on the left, and as soon as the user removes the background color, the left space should be removed. I hope you understand my point—this would look much better.

**Comment by UsmanTechJive** [comment]
> **In addition to background colors, we should have an option to apply borders. This should include various styles like dotted, dashed, and solid lines, along with customizable border width (in pixels) and border color.**

I totally agree, blocks should have an option for borders as well. Sometimes I want to use this feature myself but, unfortunately, can’t. I don’t think it would be difficult to add this small feature enhancement.

**Comment by Hzz-0802** [comment]
plugin **custom-block**，With Border (outline)
![2024-10-31_214909](https://github.com/user-attachments/assets/622e7914-c468-41b4-8e85-f1746de826a5)
![2024-10-31_215136](https://github.com/user-attachments/assets/87d3b968-16af-4a83-afc0-f8b0519bed50)

**Comment by TCOTC** [comment]
As far as I know of notes app, there is no function to add a border to the block, and I don't think SiYuan will add this function.

Currently, in Siyuan you can achieve this functionality by using code snippets and custom attributes:

```css
.protyle-wysiwyg [data-node-id][custom-box-shadow="true"] {
	box-shadow: inset 0 0 0 2px #f5822e;
}
```

![image](https://github.com/user-attachments/assets/21a5b49e-44a5-444d-925b-506a7b51303b)

![image](https://github.com/user-attachments/assets/13963a6b-cbe8-49bc-8f03-16d19d576bab)

**Comment by itszarnish** [comment]
If adding borders to blocks isn't possible, then at least the Custom Attribute UI could be improved. 

For example, instead of having to create box-shadow or other CSS styles for specific blocks repeatedly, there could be a way to set these styles once. Maybe we could have a section in CSS Snippets or a separate Custom menu in Settings where we can define styles like box-shadow. 

Once added, these predefined styles could appear as options in the custom settings for each block, allowing us to easily click and apply them as needed. I think this could be a practical solution if adding borders directly isn't feasible.

Please think about it; it would definitely speed up the process.

**Comment by itszarnish** [comment]
Also, am I getting this right? It's not applying the styles.


![image](https://github.com/user-attachments/assets/c4fa107f-4cb9-4072-97d1-535861e74257)

**Comment by TCOTC** [comment]
Refer to my operation. And I think the sy-quick-attr plugin is very convenient:

```css
.protyle-wysiwyg [data-node-id][custom-box-shadow="true"] {
	box-shadow: inset 0 0 0 2px #f5822e;
}
```

```
{
    "Border": {
        "box-shadow": "true"
    },
    "Cancel Border": {
        "box-shadow": ""
    }
}
```


https://github.com/user-attachments/assets/eda9d305-21c6-4ab3-81c8-45c2bf13fe44

**Comment by 88250** [comment]
Thank you for your discussion. We will close the issue first.

**Comment by itszarnish** [comment]
> Refer to my operation. And I think the sy-quick-attr plugin is very convenient:
> 
> ```css
> .protyle-wysiwyg [data-node-id][custom-box-shadow="true"] {
> 	box-shadow: inset 0 0 0 2px #f5822e;
> }
> ```
> 
> ```
> {
>     "Border": {
>         "box-shadow": "true"
>     },
>     "Cancel Border": {
>         "box-shadow": ""
>     }
> }
> ```
> 
>  video.mp4

Thanks, this is useful. I appreciate it.

**Comment by itszarnish** [comment]
> **I think if the user hasn't applied any background color, it looks okay. However, once a background color is applied, there should be space on the left to match the top, right, and bottom spaces, so the content looks more balanced. I believe this can be improved.**
> 

Would you please fix this issue?

![image](https://github.com/user-attachments/assets/d8aae4a1-e0d6-4eec-9c4e-1dbe0ba1c35b)

**Comment by TCOTC** [comment]
Actually this isn't a problem that needs to be fixed, the Super block is designed to have no left and right padding to keep the text aligned:

![image](https://github.com/user-attachments/assets/650cf4af-59ae-41aa-9178-ce93585c126f)

**Comment by itszarnish** [comment]
Appreciated.
